### DISCUSSION TITLE
Is there a way to annotate string as unocss class?

### DISCUSSION THREAD

**Discussion by DrSensor** [unocss/unocss#1009] [Q&A]
Here is my use case
```jsx
let isPlay = false
const [animation, animate] = createSignal(/*@uno animate*/"pulse count-1 duration-2s")
    , togglePlayer = () =>
        (isPlay = !isPlay)
          ? animate(/*@uno animate*/"bounce-alt")
          : animate("")

return <button css:animate={animation} onclick={togglePlayer} />
```
Or maybe using tagged template string
```jsx
import uno from "uno.css"

let isPlay = false
const [animation, animate] = createSignal(uno.animate`pulse count-1 duration-2s`)
    , togglePlayer = () =>
        (isPlay = !isPlay)
          ? animate(uno`animate-bounce-alt`)
          : animate("")

return <button css:animate={animation} onclick={togglePlayer} />
```

**Comment by zyyv** [comment]
How about use [transformer-variant-group](https://github.com/unocss/unocss/blob/main/packages/transformer-variant-group/README.md)🤔️
Like this [playground](https://unocss.antfu.me/play/?html=DwEwlgbgBAxgNgQwM5ILwCIEDswFsEAuApgLQAUARgPYCuWMpCcBstWBJAjAAxQg0AnQmCpYuSAJToAfACgoUOlRgpZwAPTgIcoA&config=JYWwDg9gTgLgBAbwFBzgEwKYDNgDsMDCEuOA5gDQpxhQYDOGMAgjDFMAEYCuMwWAnpVQ16jAJIBjYnSHVaDGAFVcEWWwCGuOlmggMUAGrr2mmAHEoELmCQBfOFksg4Aci4qJdOi6RIMAD0hYdGx1LgAbeEwcfCISYFIACmRUKAj6AC44AG0qVGyXCS46GAgQAFo08IwXckQ4KXDoLJdaNBc4WwBdKi7ZOgALaBgimDoslNRXIpKy8sHh0ZcWmACYcvDSOFX-dehNUgw4IYA3fQyd9dX1cJ9UW1kRBXGcvLlRJRVEgEpZYXlGCw2JweHx%2BD8-u8FJJpMk3qg6BIbhgsgBGAB0ACZIagJGhcC0BqwwOMAPSk%2BggdGDUl3Ka2X69NRQTTaXT6F65KbbFlaHRQPSGYzAUwWKxgCFMuzfJBAA&options=N4IgzgLgTglgxhEAuaBXApgXyA)

**Reply by DrSensor** [reply]
How to change the class at runtime?
For example, changing from unocss `animate-bounce` to `animate-pulse` when a button is clicked.
